azure-pipelines.yml

```diff
@@ -43,9 +47,24 @@ steps:
 - checkout: self
   clean: boolean
   submodules: recursive
+- task: CMake@1
+  inputs:
+    workingDir: 'build'
+    cmakeArgs: '$(CMAKE_EXTRA_FLAGS) ../src'
+  condition: eq( variables['Agent.OS'], 'Darwin')
+- script: |
+    cd build
+    make
+  displayName: '$(TEST_TARGET) Build & Test'
+  condition: eq( variables['Agent.OS'], 'Darwin')
+- script: |
+    cd build
+    ctest -T test --output-on-failure -V
+  displayName: '$(TEST_TARGET) Build & Test'
+  condition: eq( variables['Agent.OS'], 'Darwin')
```

Comment:
These condition variables add steps to the pipeline that get skipped 
-- e.g. https://dev.azure.com/axom/axom/_build/results?buildId=72

Everything appears to be working, but it took a little digging to figure out why 1 or 2 of 3 `Build and Test` steps were skipped.

Reply:
it would be good to know if there is a clean way to partition the steps (other than using your own scripts) in support of the matrix.

This seems to be the drag back to the some of the yaml configs.

Reply:
looks like you can include yaml templates for sub cases?

https://github.com/pytorch/builder/tree/master/windows


Their azure dev ops page isn't public, so I am not sure what this all looks like when run.

Reply:
templates look promising: 

https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#template-references

Reply:
@cyrush and @kennyweiss are you guys OK with getting this in now and we can improve it later when we understand better ways to do this?

Reply:
yes

Reply:
I saw the skipped steps, but wasn't sure what they were telling me. I was planning to discuss this with @davidbeckingsale we he gets back from travel.

Reply:
I put some debugging 'echo' statements in and it looked like all step were run. Maybe, it's just a matter of removing the display name business....